seed/seedwriter/writer.go

```diff
@@ -733,15 +750,55 @@ func (w *Writer) SetInfo(sn *SeedSnap, info *snap.Info) error {
 
 	if sn.local {
 		sn.SnapRef = info
-		// nothing more to do
-		return nil
+		return w.assignLocalComponents(sn)
 	}
 
 	p, err := w.tree.snapPath(sn)
 	if err != nil {
 		return err
 	}
 	sn.Path = p
+
+	return nil
+}
+
+type ByCompName []SeedComponent
```

Comment:
I don't think this needs to be exported.

Reply:
Good catch, unexported